[voq/inbandif] Voq inbandif port

vganesan-nokia opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines:
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "closes #xxxx",
"fixes #xxxx" or "resolves #xxxx" so that GitHub automatically closes the related
issue when the PR is merged.

If you are adding/modifying/removing any command or utility script, please also
make sure to add/modify/remove any unit tests from the tests
directory as appropriate.

If you are modifying or removing an existing 'show', 'config' or 'sonic-clear'
subcommand, or you are adding a new subcommand, please make sure you also
update the Command Line Reference Guide (doc/Command-Reference.md) to reflect
your changes.

Please provide the following information:
-->

**- What I did**

Inband port can be made available in PORT table. But regular port handlngs are
not applicable for Inband port. This PR has change to avoid regular port handling for inband port

Ref: VOQ HLD https://github.com/Azure/SONiC/blob/master/doc/voq/voq_hld.md

**- How I did it**

The inband port is considered similar to back plane port. An inband_prefix = "Ethernet-IB" and this is checked in all places to avoid regular port handling.

**- How to verify it**

- Expose a port with name prefixed with "Ethernet-IB" to SONiC via port_config.ini
- Have an entry in the PORT table for this port with name prefixed with "Ethernet-IB"
- When the system is booted, there should not be errors related to Inband interface

**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**

Note: This PR depends on sonic-buildimage PR https://github.com/Azure/sonic-buildimage/pull/6477

anshuv-mfst commented:
@eswaran (Arista), @abdosi  - could you please review the PR, Thanks.

ngoc-do commented on scripts/sfpshow:
```diff
@@ -322,7 +322,7 @@ class SFPShow(object):
             for i in sorted_table_keys:
                 interface = re.split(':', i, maxsplit=1)[-1].strip()
-                if interface and interface.startswith(front_panel_prefix()) and not interface.startswith(backplane_prefix()):
+                if interface and interface.startswith(front_panel_prefix()) and not interface.startswith(backplane_prefix()) and not interface.startswith(inband_prefix()):
```
I think right now, there is no need to check `not interface.startswith(backplane_prefix()) and not interface.startswith(inband_prefix())` since `front_panel_prefix` is different from the other prefixes. But it's safer in case in future the prefixes are changed. So approved.

vganesan-nokia replied:
Thanks for approving. But we changed the prefix. As per the review comments during Recycle port HLD, we changed the Inband port prefix to "Ethernet-IB". With this, it is required to handle inband_prefix similar to backplane_prefix.

abdosi reviewed (dismissed):
LGTM

abdosi commented:
/Azurepipelines run

anshuv-mfst commented:
@ngoc-do - please re-approve

anshuv-mfst commented:
@ngoc-do, @eswaranb - please re-approve.

abdosi commented:
can we add unit test case to check port is skip.

vganesan-nokia commented:
> can we add unit test case to check port is skip.

The existing system neigh test has been enhanced to add inband port if. Without inband interface, the neighbor will not be synced.  So in virtual chassis setup I use one of the front panel ports as the inband port. The system neigh test implicitly tests for inband if configuration, oper up and host if creation for the inband.

vganesan-nokia commented:
> can we add unit test case to check port is skip.

Yes. I'll add test case to check skipping Ethernet-IB.

vganesan-nokia commented:
Capturing approvals before rebasing

![image](https://user-images.githubusercontent.com/67648637/113929381-02b61580-97be-11eb-9b45-819145c5db3d.png)

vganesan-nokia commented:
> > can we add unit test case to check port is skip.
> 
> Yes. I'll add test case to check skipping Ethernet-IB.

Fixed. Added test case for route check to unit test changes done in route_check.py. Updated the mock appl db to include inband interface so that the code changes in sfpshow will be unit tested (the verification of skipping inband port).

renukamanavalan commented:
@abdosi, should this get into 202012 ?
